docs/csharp/csharp-7.md

```diff
@@ -16,64 +16,539 @@ ms.assetid: fd41596d-d0c2-4816-b94d-c4d00a5d0243
+<!-- * [`throw` Expressions](#throw-expressions) -->
 * [Generalized async return types](#generalized-async-return-types)
 * [Numeric literal syntax improvements](#numeric-literal-syntax-improvements)
 
-Two of the most interesting features don't make that list. The first
-is the shortened release cycle. C# 7 is following C# 6 much more quickly.
-The second is that C# 7 has [features contributed by the community](#expression-bodied-everything-preview-5), not
-the C# compiler team. The language is truly open.
-
-The remainder of this topic provides an overview 
+The remainder of this topic discusses each of the features. For each feature,
+you'll learn the reasoning behind it. You'll learn the syntax. You'll see
+some sample scenarios where using the new feature will make you more 
+productive as a developer. 
 
 ## `out` variables
 
+The existing syntax that supports `out` parameter has been improved
+in this version.  
+
+You can now declare `out` variables in the argument list of a method call,
+rather than writing a separate declaration statement:
+
+[!code-csharp[OutVariableDeclarations](../../samples/snippets/csharp/new-in-7/new-in-7/program.cs#01_OutVariableDeclarations "Out variable declarations")]
+
+You may want to specify the type of the `out` variable for clarity,
+as shown above. However, the language does support using an implicitly
+typed local variable:
+
+[!code-csharp[OutVarVariableDeclarations](../../samples/snippets/csharp/new-in-7/new-in-7/program.cs#02_OutVarVariableDeclarations "Implicitly typed Out variable")]
+
+<!--
+Add a sample at RC that shows how if statements
+scope out variables.
+-->
+
+Previously, you would need to separate the declaration of the out variable
+and its initialization into two different statements:
+
+[!code-csharp[OutVariableOldStyle](../../samples/snippets/csharp/new-in-7/new-in-7/program.cs#03_OutVariableOldStyle "classic out variable declaration")]
+
+This small language change improves your productivity in a couple ways.
+First, the code is easier to read. You declare the out variable where you
+use it, not on another line above.
+Second, by declaring the `out` variable where it is used in a method call,
+you don't need to assign an initial value, and you can't accidentally use
+it before it is assigned.
+
+The scope of the variable declaration minimizes the chance for mis-using
+the variable as well.  The `out` variable is scoped to the enclosing block.
+By minimizing the scope, you avoid name collisions with other variables and
+you minimize the chance of misuse.
+
+The most common use for this feature will be the `TryParse` pattern. In this
+pattern, a method returns a `bool` indicating success or failure, and an
+`out` variable that provides the result, in cases when the method succeeds.
 
 ## Tuples
 
-### Named and unnamed Tuples
+The syntax used for classes and structs provides a rich syntax to explain
```

Comment:
The syntax...provides a rich syntax is awkward. Perhaps, "C# provides a rich syntax for classes and structs to explain..."